javascript/javascript1/week2/homework5.js

```diff
@@ -30,13 +30,21 @@ const candyPriceList = {
   "chewing-gum": 0.03,
 };
 
+// const addCandy = (candyType, weight) => {
+//   for (let candy in candyPriceList) {
+//     if (candy === candyType) {
+//       boughtCandyPrices.push(weight * candyPriceList[candy]);
+//       break;
+//     }
+//   }
+// };
+//IMPROVED LOGIC TO LOOK INSIDE AN OBJECT
 const addCandy = (candyType, weight) => {
-  for (let candy in candyPriceList) {
-    if (candy === candyType) {
-      boughtCandyPrices.push(weight * candyPriceList[candy]);
-      break; // Exit the loop once the matching type is found
-    }
-  }
+  Object.entries(candyPriceList).find(([value]) =>
+    value === candyType
+      ? boughtCandyPrices.push(weight * candyPriceList[value])
+      : ""
+  );
```

Comment:
Keep it coming with taking on the challenges!

Now we're getting into much more advanced areas than the looping we were supposed to do. A smaller adjustment and that is you want to check the `key` against the `candyType`. 
`Object.entries(candyPriceList).find(([key, value]) =>  key === candyType...`

Also, as a general rule of thumb is that when you see `""` as part of the `else` statement then you can shorten it.

You can actually write:
`key === candyType && boughtCandyPrices.push(weight * candyPriceList[value])`